### PR TITLE
email send fail

### PR DESCRIPTION
**Pull request by priyankadhiman3006**

### CODE REVIEW

File: app/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailService.scala [hunk 1]
```diff
@@ -60,10 +60,7 @@ class SendEmailServiceImpl @Inject() (emailSendConnector: SendEmailConnector, co
     result.subflatMap { response =>
       response.status match {
         case ACCEPTED => Right(EmailSendResult.EmailSent)
-        case other    =>
-          Left(
-            Error(s"Call to sendEmail came back with status $other")
-          )
+        case _        => Right(EmailSendResult.EmailSentFailure)
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailService.scala:63]: it'd be nice to have a log message here saying what the status was - we lose that information after this point

File: test/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailServiceImplSpec.scala [hunk 2]
```diff
@@ -167,6 +149,38 @@ class SendEmailServiceImplSpec extends AnyWordSpec with Matchers with MockFactor
 
         }
 
+        "request json is not parsed and email is not send" when {
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailServiceImplSpec.scala:152]: this sits at the same level as `"return successfully" when` I think so it'd make more sense like:
```
"return an EmailSentFailure" when { ... }
```

File: test/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailServiceImplSpec.scala [hunk 3]
```diff
@@ -167,6 +149,38 @@ class SendEmailServiceImplSpec extends AnyWordSpec with Matchers with MockFactor
...
+
+          val emailSendRequest                                                                = EmailSendRequest(List(emailAddress), "template_EN", emailParameter)
+          val authenticatedRequest                                                            = AuthenticatedRequest(
+            new MessagesRequest(FakeRequest().withCookies(Cookie("PLAY_LANG", "en")), messagesApi)
+          )
+          implicit val requestWithSessionData: RequestWithSessionData[AnyContentAsEmpty.type] =
+            RequestWithSessionData(authenticatedRequest, session)
+
+          def emailSendFailure() = {
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailServiceImplSpec.scala:161]: maybe something like `testIsEmailSendFailure` is more descriptive?

Comment (priyankadhiman3006) [reply test/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailServiceImplSpec.scala]: [updated](https://github.com/hmrc/hec-applicant-frontend/pull/125/commits/a7bd7b99180cc41e23e13a2b62a1f713040ead94)

File: app/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailService.scala (hunk 1 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailService.scala]: [updated](https://github.com/hmrc/hec-applicant-frontend/pull/125/commits/a7bd7b99180cc41e23e13a2b62a1f713040ead94)

File: app/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailService.scala [hunk 4]
```diff
@@ -60,7 +62,9 @@ class SendEmailServiceImpl @Inject() (emailSendConnector: SendEmailConnector, co
     result.subflatMap { response =>
       response.status match {
         case ACCEPTED => Right(EmailSendResult.EmailSent)
-        case _        => Right(EmailSendResult.EmailSentFailure)
+        case other    =>
+          logger.info(s"Response for send email call came back with status : $other")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/services/SendEmailService.scala:66]: `warn` would be a more appropriate log level here

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**